core/lib/zksync_core/src/lib.rs

```diff
@@ -723,38 +805,86 @@ pub async fn initialize_components(
             .default_priority_fee_per_gas;
         let l1_chain_id = genesis_config.l1_chain_id;
 
-        let eth_client = PKSigningClient::new_raw(
-            operator_private_key.clone(),
-            diamond_proxy_addr,
-            default_priority_fee_per_gas,
-            l1_chain_id,
-            query_client.clone(),
-        );
+        let gkms_op_key_name = std::env::var("GOOGLE_KMS_OP_KEY_NAME").ok();
+        let use_gkms_signing_client = gkms_op_key_name.is_some();
 
-        let eth_client_blobs = if let Some(blob_operator) = eth_sender_wallets.blob_operator {
-            let operator_blob_private_key = blob_operator.private_key().clone();
-            let client = Box::new(PKSigningClient::new_raw(
-                operator_blob_private_key,
+        let eth_tx_manager_actor;
+        if use_gkms_signing_client {
+            let eth_client = GKMSSigningClient::new_raw(
                 diamond_proxy_addr,
                 default_priority_fee_per_gas,
                 l1_chain_id,
-                query_client,
-            ));
-            Some(client as Box<dyn BoundEthInterface>)
+                query_client.clone(),
+                gkms_op_key_name
+                    .expect("gkms_op_key_name is required but was None")
+                    .to_string(),
+            )
+            .await;
+
+            let gkms_op_blob_key_name = std::env::var("GOOGLE_KMS_OP_BLOB_KEY_NAME").ok();
+            let eth_client_blobs = if gkms_op_blob_key_name.is_some() {
+                let client = Box::new(
+                    GKMSSigningClient::new_raw(
+                        diamond_proxy_addr,
+                        default_priority_fee_per_gas,
+                        l1_chain_id,
+                        query_client.clone(),
+                        gkms_op_blob_key_name
+                            .expect("gkms_op_blob_key_name is required but was None")
+                            .to_string(),
+                    )
+                    .await,
+                );
+                Some(client as Box<dyn BoundEthInterface>)
+            } else {
+                None
+            };
+
```

Comment:
Extract `gkms_op_blob_key_name` in the let clause instead?

Pseudo code only
```suggestion
            let eth_client_blobs = let Some(key_name) =  gkms_op_blob_key_name {
                let client = Box::new(
                    GKMSSigningClient::new_raw(
                        diamond_proxy_addr,
                        default_priority_fee_per_gas,
                        l1_chain_id,
                        query_client.clone(),
                        key_name.to_string(),
                    )
                    .await,
                );
                Some(client as Box<dyn BoundEthInterface>)
            } else {
                None
            };

```

Reply:
done!